src/main/java/core/basesyntax/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import java.util.NoSuchElementException;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void processOperation(FruitTransaction fruitTransaction) {
+        String fruit = fruitTransaction.getFruit();
+        int fruitsInStorage = Storage.storage.get(fruit);
+        int fruitsToBuy = fruitTransaction.getQuantity();
+        if (fruitsToBuy > fruitsInStorage) {
+            throw new NoSuchElementException("Not enough fruit for purchase. You can purchase max: "
+                    + fruitsInStorage);
+        }
+        Storage.storage.put(fruitTransaction.getFruit(), fruitsInStorage - fruitsToBuy);
```

Comment:
The implementation of the `processOperation` method does not check if the fruit exists in the `Storage` before attempting to get its quantity. This may lead to a `NullPointerException` if the fruit is not present in the storage. Consider checking if the storage contains the key before getting its value.